File: frontend/GameHubFront/src/Games/batalhaRural/services/PositionService.ts
import { Board } from "../entities/Board";
import { Axis } from "../enums/Axis";
import { Direction } from "../enums/Direction";
import { TokenInfo } from "../helpers/TokenInfo";
import { BoundRuler } from "./rules/BoundRuler";
import { CollisionRuler } from "./rules/CollisionRuler";

export class PositionService {

    static canPlaceToken(tokenInfo: TokenInfo, board: Board): boolean {

        if (BoundRuler.isOutOfBounds(tokenInfo, board) || CollisionRuler.isTileOccupied(tokenInfo, board)) return false;

        return true;
    }

    static isTokenPathOccupied(tokenInfo: TokenInfo, board: Board, axis: Axis, direction: Direction): boolean {
        switch (axis) {
            case Axis.X:
                if (!CollisionRuler.isTokenHorizontallyOverlaping(tokenInfo, board, direction)) return false;
                break;
            case Axis.Y:
                if (!CollisionRuler.isTokenVerticallyOverlaping(tokenInfo, board, direction)) return false;
                break;               
        }
        
        return true;
    }
}

File: backend/gameHubBack/services/BatalhaRural/PositionService.cs
using gameHubBack.Enums.BatalhaRural;
using gameHubBack.Services.BatalhaRural.Rules;

namespace gameHubBack.Services.BatalhaRural;

public static class PositionService
{
    public static bool CanPlaceToken(TokenInfo tokenInfo, string[][] tiles)
    {
        if (BoundRuler.IsOutOfBounds(tokenInfo, tiles) || CollisionRuler.IsTileOccupied(tokenInfo, tiles)) return false;

        return true;
    }

    public static bool IsTokenPathOccupied(TokenInfo tokenInfo, string[][] tiles, Axis axis, Direction direction)
    {
        return axis switch
        {
            Axis.X => CollisionRuler.IsTokenHorizontallyOverlapping(tokenInfo, tiles, direction),
            Axis.Y => CollisionRuler.IsTokenVerticallyOverlapping(tokenInfo, tiles, direction),
            _ => true
        };
    }
}